npm/webpack-dev-server/src/helpers/sourceRelativeWebpackModules.ts

```diff
@@ -56,7 +56,18 @@ export const cypressWebpackPath = (config: WebpackDevServerConfig) => {
   })
 }
 
-// Source the users framework from the provided projectRoot. The framework, if available, will server
+type FrameworkWebpackMapper = { [Property in typeof ALL_FRAMEWORKS[number]]: string | undefined }
```

Comment:
We could define this as `type FrameworkWebpackMapper = { [Property in typeof ALL_FRAMEWORKS[number]]?: string }` so the `undefined` entries aren't necessary for react & vue.

That said, I suppose it might be desired to have a dev actually look and think about this logic when we add new frameworks, so having the TS compiler complain if they don't explicitly add an entry for it might be good?

Reply:
That's what I was going for it. Whenever a new framework is added, this will enforce that whoever is adding the new framework considers how to properly link the framework name to the package that contains the relevant deps.